php/ext/google/protobuf/message.c

```diff
@@ -108,6 +108,76 @@ static const upb_fielddef *get_field(Message *msg, zval *member) {
   return f;
 }
 
+/**
+ * Message_has_property()
+ *
+ * Object handler for testing whether a property exists. Called when PHP code
+ * does any of:
+ *
+ *   isset($message->foobar);
+ *   property_exists($message->foobar);
+ *
+ * Note that all properties of generated messages are private, so this should
+ * only be possible to invoke from generated code, which has accessors like this
+ * (if the field has presence):
+ *
+ *   public function hasOptionalInt32()
+ *   {
+ *       return isset($this->optional_int32);
+ *   }
+ */
+static int Message_has_property(zval *obj, zval *member, int has_set_exists,
+                                void **cache_slot) {
+  Message* intern = (Message*)Z_OBJ_P(obj);
+  const upb_fielddef *f = get_field(intern, member);
+
+  if (!f) return 0;
+
+  if (!upb_fielddef_haspresence(f)) {
+    zend_throw_exception_ex(
```

Comment:
Should this be an internal error instead of exception?

Reply:
I spoke with @bshaffer and it sounds like it's better to use exceptions than errors in PHP. In fact we were talking about replacing all errors with exceptions for 4.0, since the major version will let us break backward compatibility for this.

Reply:
I think I might file an issue for this.

Reply:
Only our generated code assess this method. Seems user should never see this exception.